### ISSUE TITLE
Radio Group - Selected item is not selected in High Contrast mode

### ISSUE THREAD

**Issue by mpriour** [Esri/calcite-components#4873]
### Summary

The lack of visual information for selected items was identified as a level 3 severity issue (missing critical information) in the [High Contrast Audit](https://github.com/Esri/calcite-components/wiki/High-Contrast-Audit#radiogroup). The problem is that there is no indication of what item is selected in a radio group component when using High Contrast mode.

### Actual Behavior

High Contrast Radio Group with selected items using both solid and outline form with and without icons

<img height=400 src="https://user-images.githubusercontent.com/142636/149631506-ada043e6-b991-4764-bc4e-0da2eb0dc975.png">

### Expected Behavior

Selected item in the Radio Group should be visually obvious even in High Contrast mode.

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/radio-group/

### Reproduction Steps

Windows
1. Set Windows color theme to a high-contrast variant.
2. Open the reproduction sample (https://developers.arcgis.com/calcite-design-system/components/radio-group/) in any browser.
3. Observe the issue.

Mac
1. Open the reproduction sample (https://developers.arcgis.com/calcite-design-system/components/radio-group/) in Chrome or Edge.
2. Open Dev Tools.
3. Click on Elements tab.
4. Next to the console sub tab, there should be a "Rendering" tab (if not you will need to enable it)
5. On the Rendering tab choose Forced-colors: active from the "Emulate forced colors..." selector
6. Observe the issue.

### Reproduction Version

latest

**Comment by geospatialem** [comment]
Verified on next ✔️ 
![high contrast verification](https://user-images.githubusercontent.com/5023024/178078139-68dfbaf9-94e3-4816-b6ac-d9bb9ae781b7.png)
